src/fn_call.rs

```diff
@@ -149,11 +149,19 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a + 'mir>: crate::MiriEvalContextExt<'
         let link_name = link_name.get().trim_end_matches("$UNIX2003");
         let tcx = &{this.tcx.tcx};
 
-        // First: functions that could diverge.
+        // First: functions that diverge.
         match link_name {
             "__rust_start_panic" | "panic_impl" => {
                 return err!(MachineError("the evaluated program panicked".to_string()));
             }
+            "exit" => {
+                let code = this.read_scalar(args[0])?.to_i32()?;
+                return err!(Exit(code));
+            }
+            "ExitProcess" => {
+                let code = this.read_scalar(args[0])?.to_u32()?;
```

Comment:
even if `ExitProcess` has a `u32` return type, if we're just casting to `i32` immediately we can use `to_i32` here and deduplicate the code between `exit` and `ExitProcess`

Reply:
Done.